### PR TITLE
Admin Manage Accounts Tab

### PR DESCRIPTION
**Pull request by aksharans**
### Administrative Info
Monday Board ID: 853642866
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Manage accounts page with ability to add, delete, and edit admins
- Admin card component
- add admin routes to api

### Testing
How did you confirm your changes worked? 
- Tested adding, deleting, editing admins with backend locally, and with test data

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
- cd web, npm start
- cd backend, make, DISABLE_AUTH=true ./words-alive
- curl -H "Authorization: Bearer test-token-1" localhost:8080/admins

### CODE REVIEW

File: web/src/App.css [hunk 1]
```diff
@@ -15,6 +15,11 @@ body {
     margin-top: 10px;
 }
 
+.h1 {
```
Comment (dacruz21) [inline web/src/App.css:18]: Can you plz re-indent this file consistently?

File: web/src/api/WordsAliveAPI.tsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 // Class to encapsulate the handler for the Words Alive API
 class WordsAliveAPI {
   client: AxiosInstance;
 
   constructor(baseURL: string) {
-    this.client = axios.create({ baseURL: baseURL });
+    this.client = axios.create({ baseURL: baseURL, headers: { Authorization: 'Bearer test-token-1' } });
   }
+  // curl -H "Authorization: Bearer test-token-1" localhost:8080/admins
+  // make & DISABLE_AUTH=true ./words-alive
```
Comment (dacruz21) [inline web/src/api/WordsAliveAPI.tsx:14]: Go ahead and remove this now, we'll do this properly in the login site PR

Comment (dacruz21) [reply web/src/api/WordsAliveAPI.tsx:14]: (and remove the notes to self :slightly_smiling_face: )

File: web/src/pages/ManageAccountsPage.module.css [hunk 3]
```diff
@@ -0,0 +1,216 @@
+p {
+    margin: 0px;
+}
```
Comment (dacruz21) [inline web/src/pages/ManageAccountsPage.module.css:3]: Sorry, only class selectors in module files please

File: web/src/pages/ManageAccountsPage.module.css [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    font-size: 16px;
+    font-family: 'GothamRounded - Light';
+    -webkit-appearance: none;
+}
+
+input[type=checkbox] {
+    width: 30px;
+    height: 30px;
+    border-radius: 5px;
+}
```
Comment (dacruz21) [inline web/src/pages/ManageAccountsPage.module.css:191]: Same here, you can define a class for these inputs or use a child selector on another class

File: web/src/pages/ManageAccountsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+import { APIContext } from '../context/APIContext';
+import { Admin, CreateAdmin } from '../models/Admin';
+import AddIcon from '../assets/images/plus-circle-solid-green.svg';
+import CancelIcon from '../assets/images/times-solid.svg';
+
+import '../App.css';
+import styles from './ManageAccountsPage.module.css';
+
+
+const test: Admin[] = [
```
Comment (dacruz21) [inline web/src/pages/ManageAccountsPage.tsx:12]: Remove plz

File: web/src/components/AdminCard.module.css [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    backdrop-filter: blur(5px) brightness(68%);
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+
+.modalContentManage {
+    padding: 60px;
+    width: 50%;
+    height: 80%;
```
Comment (dacruz21) [inline web/src/components/AdminCard.module.css:68]: Can you remove the height:80% please? It looks funny on my screen but it looks totally fine if we just let it shrinkwrap.

**Review by typokign — CHANGES_REQUESTED** [review]
Ik you're still working on the checkboxes so no comment there :)

Looks fantastic, thanks for putting this together so quickly and cleanly. Just a few comments:

**Comment by aksharans** [comment]
Made the custom checkboxes similar to the upload books page (without an image). If everything looks good, I can go ahead and delete the WordsAliveAPI stuff and push again.
Also, I saw a type selector in UploadBooksPage.module.css, but I didn't have the updated version so I wasn't able to fix it in this branch

**Comment by dacruz21** [comment]
Looks good, can you try to line these labels up though?

image

**Comment by aksharans** [comment]
By line up, did you mean have everything left-aligned?

**Comment by dacruz21** [comment]
Yeah, I think it would look best if we left align the text and right align the check boxes

**Review by typokign — DISMISSED** [review]
LGTM, great work. Please ping someone else for review.

**Review by typokign — DISMISSED** [review]

**Review by typokign — DISMISSED** [review]

File: web/src/pages/ManageAccountsPage.module.css [hunk 7]
```diff
@@ -0,0 +1,239 @@
+.title {
+    composes: h2 from '../App.css';
+    text-align: left;
+    margin-top: 94px;
+    margin-left: 12px;
```
Comment (allisonbhavsar) [inline web/src/pages/ManageAccountsPage.module.css:5]: Any reason for the left margin here? In the mockup the title looks aligned all the way to the left

**Review by allisonbhavsar — CHANGES_REQUESTED** [review]
Minor style fix but other than that looks and works great! Are we leaving the error messages as alerts?

**Review by typokign — APPROVED** [review]

**Comment by aksharans** [comment]
Removed the margin left and I believe the errors messages would be alerts, @dacruz21 ?

**Comment by dacruz21** [comment]
Yeah alerts are probably fine for now, I can ask design their thoughts

**Review by allisonbhavsar — APPROVED** [review]
LGTM

**Pull request merged**